homework10/src/ba/bitcamp/homework10/task02/CollectionLinkedList.java

```diff
@@ -0,0 +1,192 @@
+package ba.bitcamp.homework10.task02;
+
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Iterator;
+
+public class CollectionLinkedList implements Collection<Integer> {
+	
+	Node start;
+	Node finish;
+	
+	public CollectionLinkedList () {
+		start = null;
+		finish = null;
+	}
+	
+	public String toString() {
+		if (start == null) {
+			return "Linked List is Empty";
+		}
+		return start.toString();
+	}
+	
+	public class Node {
+		Node next;
+		Integer value;
+		
+		public Node (Integer value) {
+			this.value = value;
+		}
+		
+		public Node getNext() {
+			return next;
+		}
+		
+		public void setNext (Node next) {
+			this.next = next;
+		}
+		
+		public Integer getValue() {
+			return value;
+		}
+		
+		public String toString() {
+			if (next == null) {
+				return value + "";
+			}
+			return value + ", " + next.toString();
+		}
+	}
+
+	public static void main(String[] args) {
+		// TODO Auto-generated method stub
+		
+		CollectionLinkedList list = new CollectionLinkedList();
+		System.out.println("Add: " + list.add(3));
+		list.add(6);
+		list.add(2);
+		list.add(5);
+		System.out.println(list);
+//		list.clear();
+		System.out.println("Contains: " + list.contains(2));
+		System.out.println("Is Empty: " + list.isEmpty());
+//		list.remove(6);
+//		list.remove(2);
+		System.out.println(list.size());
+		Integer[] array = (Integer[]) list.toArray();
+		System.out.println(Arrays.toString(array));
+		
+		
+
+	}
+
+	@Override
+	public boolean add(Integer e) {
+		Node newNode = new Node(e);
+		if (start == null) {
+			start = newNode;
+			finish = newNode;
+		} else if (start == finish) {
+			start.setNext(newNode);
+			finish = newNode;
+		} else {
+			finish.setNext(newNode);
+			finish = newNode;
+		}
+		return true;
+	}
+
+	@Override
+	public boolean addAll(Collection<? extends Integer> c) {
+		// TODO Auto-generated method stub
+		return false;
+	}
+
+	@Override
+	public void clear() {
+		start = null;
+	}
+
+	@Override
+	public boolean contains(Object o) {
+		Integer num = (Integer) o;
+		Node temp = start;
+		while (temp != null) {
+			if (temp.getValue() == num) {
+				return true;
+			}
+			temp = temp.getNext();
+		}
+		return false;
+	}
+
+	@Override
+	public boolean containsAll(Collection<?> c) {
+		// TODO Auto-generated method stub
+		return false;
+	}
+
+	@Override
+	public boolean isEmpty() {
+		return (start == null);
+	}
+
+	@Override
+	public Iterator<Integer> iterator() {
+		// TODO Auto-generated method stub
+		return null;
+	}
+
+	@Override
+	public boolean remove(Object o) {
+		Integer num = (Integer) o;
+		Node temp = start;
+		while (temp != null && temp.getNext().getValue() != num) {
```

Comment:
Ovaj dio koda baca NullPointerException. To je iz razloga to getNext() na zadnje elementu vraca null, a na kojem pokusavamo pozvati getValue(). Potrebno je iterirati kroz listu, pri cemu cuvamo predhodni. Nakon sto nadjemo odgovarajuci element, sacuvani predhodni iskoristiti da prespojimo listu.